pkg/ring/lifecycler.go

```diff
@@ -246,18 +248,22 @@ func (i *Lifecycler) ChangeState(ctx context.Context, state IngesterState) error
 	return <-err
 }
 
-func (i *Lifecycler) getTokens() []uint32 {
+func (i *Lifecycler) getTokens() Tokens {
 	i.stateMtx.Lock()
 	defer i.stateMtx.Unlock()
 	return i.tokens
 }
 
-func (i *Lifecycler) setTokens(tokens []uint32) {
+func (i *Lifecycler) setTokens(tokens Tokens) {
 	tokensOwned.WithLabelValues(i.RingName).Set(float64(len(tokens)))
 
 	i.stateMtx.Lock()
 	defer i.stateMtx.Unlock()
+
 	i.tokens = tokens
+	if err := i.tokens.StoreToFile(i.cfg.TokensFilePath); err != nil {
```

Comment:
I think there should be a check for `i.cfg.TokensFilePath` being non-empty here. Otherwise `StoreToFile` returns error (correct, since it cannot store to empty filename), but on this level, it's not really an error but disabled option.

Reply:
I missed it while doing the review, but Peter is right. This should be fixed.